numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -1557,7 +1557,7 @@ static PyObject *npyiter_multi_index_get(NewNpyArrayIterObject *self)
         }
         for (idim = 0; idim < ndim; ++idim) {
             PyTuple_SET_ITEM(ret, idim,
-                    PyLong_FromLong(multi_index[idim]));
+                    PyLong_FromSsize_t(multi_index[idim]));
```

Comment:
Really we should also be checking if `PyLong_FromSsize_t` returns `NULL` too, to avoid segfaults if a MemoryError occurs

Reply:
Its a bit of an odd one maybe, but might as well use `PyArray_IntTupleFromIntp` probably.

Reply:
Adopted `PyArray_IntTupleFromIntp`, which properly checks `long long` and errors